lock-api/src/main/java/com/palantir/lock/client/SharedLockToken.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.google.common.base.Preconditions;
+import com.palantir.lock.v2.LockToken;
+
+class SharedLockToken implements LockToken {
+    private final UUID requestId;
+    private final ReferenceCounter referenceCounter;
+
+    private volatile boolean unlocked;
```

Comment:
nit: since `unlocked` is only read/written during construction with assignment to a final field and then in synchronized methods, I don't think this needs to be volatile, unless I'm missing something

Reply:
I agree, this doesn't need to be volatile.